### PR TITLE
feat(errors) Expose errors through Iframe API

### PR DESCRIPTION
**Pull request by horymury**

### CODE REVIEW

File: modules/API/API.js [hunk 1]
```diff
@@ -1325,6 +1325,19 @@ class API {
...
 
+    /**
+     * Notify external application (if API is enabled) that an error was thrown.
+     *
+     * @param {Object} error - The error.
+     * @returns {void}
+     */
+    notifyErrorThrown(error: Object) {
+        this._sendEvent({
+            name: 'error-thrown',
```
Comment (saghul) [inline modules/API/API.js:1336]: can we just call this "error" ? nothing is really thrown...

File: react/features/overlay/middleware.js [hunk 2]
```diff
@@ -16,6 +20,47 @@ const NON_OVERLAY_ERRORS = [
...
+};
+
+/**
+ * Gets the error type and whether it's fatal or not.
+ *
+ * @param {Function} getState - The redux function for fetching the current state.
+ * @param {Object|string} error - The error to process.
+ * @returns {void}
+ */
+const getErrorExtraInfo = (getState, error) => {
```
Comment (saghul) [inline react/features/overlay/middleware.js:36]: I really like how you implemented this!

**Review by saghul — COMMENTED** [review]
One nit on the naming , LGTM otherwise.

File: modules/API/API.js (hunk 1 above)
Comment (horymury) [reply modules/API/API.js]: done

File: react/features/overlay/middleware.js (hunk 2 above)
Comment (horymury) [reply react/features/overlay/middleware.js:36]: thank you

**Review by saghul — APPROVED** [review]
LGTM! Can we get some docs for the handbook too please?

**Pull request merged**